### ISSUE TITLE
Opam compilation on Cygwin fails (opam-admin-cache)

### ISSUE THREAD

**Issue by MisterDA** [ocaml/opam#4090]
Hi!

I’m trying to bootstrap OCaml+Opam in Cygwin.
I’ve set up the environment as described in the readme.

```
setup-x86_64 --root=C:\cygwin64 --quiet-mode --no-desktop --no-startmenu --packages=make,mingw64-i686-gcc-core,mingw64-x86_64-gcc-core,m4,patch
```

Then ran:

```
make compiler
make lib-pkg
./configure
make 
```

But I got this error in the last step:

```
dune build --profile=release  opam-installer.install opam.install
Done: 899/902 (jobs: 1)File "doc/man/opam-admin-cache.err", line 1, characters 0-0:
        diff (internal) (exit 1)
(cd _build/default && C:\cygwin64\bin\diff.exe -u --strip-trailing-cr doc/man/opam-admin-cache.err doc/man/opam-admin-cache.0)
--- doc/man/opam-admin-cache.err        2020-02-18 18:27:56.968589100 +0100
+++ doc/man/opam-admin-cache.0  2020-02-18 18:25:52.391389600 +0100
@@ -1,6 +0,0 @@
-cmdliner error: Illegal escape char 'o' in "$(i,DIR) (absent=~\\opam-2.0.6\\_build\\default\\doc\\man\\cache)"
-cmdliner error: Illegal escape char '_' in "$(i,DIR) (absent=~\\opam-2.0.6\\_build\\default\\doc\\man\\cache)"
-cmdliner error: Illegal escape char 'd' in "$(i,DIR) (absent=~\\opam-2.0.6\\_build\\default\\doc\\man\\cache)"
-cmdliner error: Illegal escape char 'd' in "$(i,DIR) (absent=~\\opam-2.0.6\\_build\\default\\doc\\man\\cache)"
-cmdliner error: Illegal escape char 'm' in "$(i,DIR) (absent=~\\opam-2.0.6\\_build\\default\\doc\\man\\cache)"
-cmdliner error: Illegal escape char 'c' in "$(i,DIR) (absent=~\\opam-2.0.6\\_build\\default\\doc\\man\\cache)"
make: *** [Makefile:114: build-opam] Error 1
```

What can be done to fix this?

Thanks!

**Comment by rjbou** [comment]
I'll ping @dra27 and maybe @dbuenzli on this one.

**Comment by MisterDA** [comment]
This is the script I’m using to compile on Windows.
<details>

``` shell
#!/bin/sh

# setup-x86_64 --root=C:\cygwin64 --quiet-mode --no-desktop --no-startmenu --packages=make,mingw64-x86_64-gcc-core,m4,patch --site "http://mirrors.kernel.org/sourceware/cygwin/"
# C:\cygwin64\bin\mintty -
# echo "none /cygdrive cygdrive noacl,binary,posix=0,user 0 0" > /etc/fstab

OCAML_VERSION=4.09.0
OCAML_PATH=ocaml-${OCAML_VERSION}
OPAM_VERSION=2.0.6
FLEXDLL_VERSION=0.37

PREFIX=/usr/local
BUILD=x86_64-pc-cygwin
HOST=x86_64-w64-mingw32

BUILDDIR=/cygdrive/c/ocaml_platform_build

MAKEFLAGS=-j$(nproc)
export MAKEFLAGS

CYGWIN="nodosfilewarning winsymlinks:native"
export CYGWIN

mkdir -p "$BUILDDIR"
cd "$BUILDDIR" || exit 1

curl -SLfs "https://github.com/ocaml/ocaml/archive/${OCAML_VERSION}.tar.gz" -o "${OCAML_VERSION}.tar.gz"
tar xf "${OCAML_VERSION}.tar.gz"

curl -SLfs "https://github.com/alainfrisch/flexdll/archive/${FLEXDLL_VERSION}.tar.gz" -o "${FLEXDLL_VERSION}.tar.gz"
tar xf "${FLEXDLL_VERSION}.tar.gz"
mv "flexdll-${FLEXDLL_VERSION}"/* "${OCAML_PATH}/flexdll/"

cd "${OCAML_PATH}" || exit 1
./configure --build="$BUILD" --host="$HOST" --prefix="$PREFIX"
make flexdll
make world.opt
make flexlink.opt
make install

cd "$BUILDDIR" || exit 1

curl -SLfs "https://github.com/ocaml/opam/releases/download/${OPAM_VERSION}/opam-full-${OPAM_VERSION}.tar.gz" -o "opam-full-${OPAM_VERSION}.tar.gz"
tar xf "opam-full-${OPAM_VERSION}.tar.gz"

OCAMLLIB="C:/cygwin64$PREFIX/lib/ocaml"
CAML_LD_LIBRARY_PATH="$OCAMLLIB/stublibs;$OCAMLLIB"
export OCAMLLIB
export CAML_LD_LIBRARY_PATH

cd "opam-full-${OPAM_VERSION}" || exit 1
./configure --build="$BUILD" --host="$HOST" --prefix="$PREFIX"
make lib-ext all -j1 OCAMLC='ocamlc -unsafe-string' OCAMLOPT='ocamlopt -unsafe-string'
make install

cd "$BUILDDIR" || exit 1

OPAMYES=true
export OPAMYES

opam --init -a --disable-sandboxing
eval $(opam env)
 ```
</details>

It still fails on the same error.

**Comment by MisterDA** [comment]
The error disappears when using master (8da623adb121a48c51c1ec1bf850fe3db8bafc68).

**Comment by rjbou** [comment]
Finally found out! It was resolved with #3911
